storage/rocksdb/ha_rocksdb.cc

```diff
@@ -10397,6 +10464,20 @@ int ha_rocksdb::inplace_populate_sk(
       }
 
       /* Create new secondary index entry */
+      m_sk_tails.clear();
+      if (index->m_total_index_flags_length > 0) {
+        m_sk_tails.allocate(0, index->m_total_index_flags_length);
+      }
+
+      /* Insert TTL timestamp */
+      if (Rdb_key_def::has_index_flag(index->m_index_flags_bitmap,
+                                      Rdb_key_def::TTL_FLAG)) {
+        uint32 ttl_offset = Rdb_key_def::calculate_index_flag_offset(
+            index->m_index_flags_bitmap, Rdb_key_def::TTL_FLAG);
+        memcpy(m_sk_tails.ptr() + ttl_offset, m_ttl_bytes,
+               ROCKSDB_SIZEOF_TTL_RECORD);
+      }
+
```

Comment:
I would suggest moving this code into a helper function. Same helper can be called from update_sk().